python/ray/tune/trainable.py

```diff
@@ -164,7 +172,11 @@ def __init__(
         self.sync_function_tpl = sync_function_tpl or self._sync_function_tpl
         self.storage_client = None
 
-        if self.uses_cloud_checkpointing:
+        if self.uses_cloud_checkpointing and self.sync_function_tpl:
+            # Keep this only for custom sync functions and
+            # backwards compatibility.
+            # Todo (krfricke): We should find a way to register custom
```

Comment:
what does this mean?

Reply:
Users can currently specify a `sync_function_tpl`. This functionality should continue to exist for backwards compatibility. There is currently no way to specify own syncers to be used with `Checkpoint`.
https://github.com/ray-project/ray/pull/23370/ is open to address this (but will liekly be refactored according to Eric's proposal), but until something like that is in place we'll ahve to keep this legacy path here